src/R/Interpreters/Impl/Implementation/RInstallation.cs

```diff
@@ -82,13 +62,22 @@ public sealed class RInstallation {
         private IEnumerable<IRInterpreterInfo> GetCompatibleEnginesFromRegistry(ISupportedRVersionRange svr) {
             svr = svr ?? new SupportedRVersionRange();
             var engines = GetInstalledEnginesFromRegistry().Where(e => svr.IsCompatibleVersion(e.Version));
-            // Remove duplicates (MRC registers under multiple keys)
-            var mrc = engines.FirstOrDefault(e => e.Name.Contains("Microsoft"));
-            if(mrc != null) {
-                var dupes = engines.Where(e => e.InstallPath.EqualsIgnoreCase(mrc.InstallPath)).Except(new IRInterpreterInfo[] { mrc });
-                engines = engines.Except(dupes);
+            // Among duplicates by path (MRC registers under multiple keys) take the highest version
+            var microsoftEngines = engines.Where(e => e.Name.Contains("Microsoft") || e.InstallPath.Contains("Microsoft"));
+            var installLocations = microsoftEngines.Select(e => e.InstallPath.TrimTrailingSlash()).Distinct();
```

Comment:
Why would we want to display duplicate CRAN entries, if two keys are registered to the same path? It would seem that single path = single workspace, regardless of which interpreter that is.

Basically, the only case this would apply is if someone does end up creating such duplicate paths. If they do, there's no harm in cleaning that up. If they don't, then none of this logic will kick in.